### PR TITLE
add extension op proto

### PR DESCRIPTION
**Pull request by wey-code**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation and Context

Add the definition of fusion operator, which is implemented in: https://github.com/DeepLink-org/DIOPI/pull/469

## Checklist
**Before PR**:

- [ ] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: proto/include/diopi/functions_ext.h [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+#endif  // __cplusplus
+
+/**
+ * @brief Apply rotary embedding operation to an input tensor.
+ * @param[in] ctx Context environment.
+ * @param[out] out The output tensor containing the rotary embeddings. type = [float32, float16, float64].
+ * @param[in] x The input tensor which rotary embedding will be applied. type = [float32, float16, float64].
+ * @param[in] cos The cosine values. type = [float32, float16, float64].
+ * @param[in] sin The sine values. type = [float32, float16, float64].
+ * @param[in] conj bool: If `false`, computes regular rotary embeddings. If `true`, computes the complex conjugate of the rotary embeddings.
```
Comment (yangbofun) [inline proto/include/diopi/functions_ext.h:23]: conj是共轭，true时，是forward， false时是backward。

**Comment by yangbofun** [comment]
指定下tensor的维度的，以及对应维度的意义

File: proto/include/diopi/functions_ext.h (hunk 1 above)
Comment (wey-code) [reply proto/include/diopi/functions_ext.h]: done

**Comment by wey-code** [comment]
> 指定下tensor的维度的，以及对应维度的意义

done

File: proto/include/diopi/functions_ext.h [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+ *   - for padded: shape = [batch_size, q_seq_len, head_num, head_dim]
+ * @param[out] softmax_lse Tensor representing the log-sum-exp of the softmax values. type = [float32, float16, float64].
+ *   - for unpaded: shape = [batch_size, head_num, max_q]
+ *   - for padded: shape = [batch_size, head_num, q_seq_len]
+ * @param[out] gen Handle for the random number generator used in dropout.
+ * @param[out] debug_attn_mask Debugging tensor for the attention mask (returned if return_debug_mask is true). type = [bool].
+ *   - for unpadded: shape = [batch_size, num_heads, max_q, max_k]
+ *   - for padded: shape = [batch_size, num_heads, q_seq_len, k_seq_len]
+ */
+DIOPI_API diopiError_t diopiMultiHeadAttention(diopiContextHandle_t ctx, diopiConstTensorHandle_t q, diopiConstTensorHandle_t k, diopiConstTensorHandle_t v,
```
Comment (yangbofun) [inline proto/include/diopi/functions_ext.h:100]: 这里拆成diopiMultiHeadAttentionVarLen 和diopiMultiHeadAttention

**Review by yangbofun — APPROVED** [review]

**Pull request merged**